File: pinyougou-user-web/src/main/webapp/js/controller/userController.js
//控制层
app.controller('userController', function ($scope, $controller, userService) {

    $scope.register = function () {
        //注册
        if ($scope.entity.password != $scope.password) {
            alert("两次输入的密码不一致，请重新输入");
            return;
        }
        if($scope.smscode==null||$scope.smscode==""){
            alert("验证码不能为空");
            return;
        }
        userService.register($scope.smscode,$scope.entity).success(
            function (response) {
                alert(response.message);
            }
        )
    }
    $scope.sendCode = function () {
        if($scope.entity.phone==null||$scope.entity.phone==''){
            alert("手机号不能为空");
            return;
        }
        userService.sendCode($scope.entity.phone).success(
            function (response) {
                alert(response.message)
            }


        )

    }


});	

File: pinyougou-user-web/src/main/java/com/pinyougou/user/controller/UserController.java
package com.pinyougou.user.controller;

import com.pinyougou.pojo.TbUser;
import com.pinyougou.utils.PhoneFormatCheckUtils;
import entity.Result;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.user.service.UserService;

/**
 * controller
 * @author Administrator
 *
 */
@RestController
@RequestMapping("/user")
public class UserController {

	@Reference
	private UserService userService;

	@RequestMapping("/register")
	public Result register(@RequestBody TbUser user,String smscode){
		try {
			if(!userService.compareCode(user.getPhone(), smscode)){
				return new Result(false,"验证码输入错误");
			}
			userService.add(user);
			return new Result(true,"注册成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false,"注册失败");

		}
	}
	@RequestMapping("/sendCode")
	public Result sendCode(String phone){
		if (!PhoneFormatCheckUtils.isChinaPhoneLegal(phone)) {
			return new Result(false,"手机号格式错误");
		}
		try {
			userService.sendCode(phone);
			return new Result(true,"发送成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false,"发送失败");
		}
	}



}
